project/src/components/catalog/catalog.tsx

```diff
@@ -1,29 +1,38 @@
 import GenresList from '../genres-list/genres-list';
 import CatalogFilmsList from '../catalog-films-list/catalog-films-list';
-import { FilmType } from '../../types/film';
 import BtnShowMore from '../btn-show-more/btn-show-more';
+import { filteringFilms } from '../../utils';
+import { useAppSelector } from '../../hooks';
+import { QuantityCards, ViewLink } from '../../const';
 
 type CatalogProps = {
-  filmCardsData: FilmType[];
   title: string;
   className?: string;
   titleClass?: string;
   isMainView?: boolean;
 };
 
-function Catalog({
-  filmCardsData,
-  title,
-  className = '',
-  titleClass = '',
-  isMainView = false,
-}: CatalogProps): JSX.Element {
+function Catalog({ title, className = '', titleClass = '', isMainView = false }: CatalogProps): JSX.Element {
+  const { films, activeGenre, activeLink, quantityShownCards } = useAppSelector((state) => state);
+  const filmsData = isMainView ? filteringFilms(films, activeGenre) : films;
```

Comment:
Функции называем с глаголом
```suggestion
  const filmsData = isMainView ? filterFilms(films, activeGenre) : films;
```